src/Window.cpp

```diff
@@ -1299,40 +1513,68 @@ void Window::updateMouseProgress(double progress) {
+            if (m_movement == Movement::RUN_RIGHT) {
+                currentTranslation += Coordinate::Cartesian( Dimensions::halfTileLength() * (      currentRotation.getSin()) * -1,  Dimensions::halfTileLength()*(1.0 - currentRotation.getCos()) * -1 );
+            }
+            else if (m_movement == Movement::RUN_LEFT) {
+                currentTranslation += Coordinate::Cartesian( Dimensions::halfTileLength() * (      currentRotation.getSin()) * +1,  Dimensions::halfTileLength()*(1.0 - currentRotation.getCos()) * +1);
+            }
+        }
+        else if (m_startingDirection == Direction::SOUTH) {
+            if (m_movement == Movement::RUN_RIGHT) {
+                currentTranslation += Coordinate::Cartesian( Dimensions::halfTileLength() * (1.0 + currentRotation.getSin()) * -1,  Dimensions::halfTileLength()*(      currentRotation.getCos()) * +1 );
+            }
+            else if (m_movement == Movement::RUN_LEFT) {
+                currentTranslation += Coordinate::Cartesian( Dimensions::halfTileLength() * (1.0 + currentRotation.getSin()) * +1,  Dimensions::halfTileLength()*(      currentRotation.getCos()) * -1);
+            }
+        }
+        else if (m_startingDirection == Direction::WEST) {
+            if (m_movement == Movement::RUN_RIGHT) {
+                currentTranslation += Coordinate::Cartesian( Dimensions::halfTileLength() * (      currentRotation.getSin()) * -1,  Dimensions::halfTileLength()*(1.0 + currentRotation.getCos()) * +1);
+            }
+            else if (m_movement == Movement::RUN_LEFT) {
+                currentTranslation += Coordinate::Cartesian( Dimensions::halfTileLength() * (      currentRotation.getSin()) * +1,  Dimensions::halfTileLength()*(1.0 + currentRotation.getCos()) * -1);
+            }
+        }else {
+            stopRun("Stop running because of a wrong direction!");
+            return;
+        }
+
+    }
+
     m_mouse->teleport(currentTranslation, currentRotation);
     if (remaining == 0.0) {
-        m_startingLocation = m_mouse->getCurrentDiscretizedTranslation();
+
+        m_startingLocation = m_mouse->getCurrentHalfDiscretizedTranslation();
         m_startingDirection = m_mouse->getCurrentDiscretizedRotation();
         m_movementProgress = 0.0;
         m_movementStepSize = 0.0;
-        if (m_movement == Movement::MOVE_FORWARD) {
+        if( (m_movement == Movement::MOVE_FULLFORWARD) || (m_movement == Movement::MOVE_HALFFORWARD) ||  (m_movement == Movement::MOVE_EDGEFORWARD) ){
             m_movesRemaining -= 1;
         }
         if (m_movesRemaining == 0) {
             m_movement = Movement::NONE;
         }
-        // determine if the goal was reached
-        if (m_maze->isInCenter(m_startingLocation)) {
-            stats->finishRun(); // record a completed start-to-finish run
-        }
-        else if (m_startingLocation.first == 0 && m_startingLocation.second == 0) {
-            stats->endUnfinishedRun();
-        }
```

Comment:
Curious: why remove this logic? Why shouldn't the simulator keep track of finished/unfinished runs?